lib/library/AssignLibraryPlugin.js

```diff
@@ -139,6 +139,11 @@ class AssignLibraryPlugin extends AbstractLibraryPlugin {
 		const result = new ConcatSource();
 		if (this.declare) {
 			const base = fullNameResolved[0];
+			if (base.search("-") !== -1) {
```

Comment:
it should be better check here, we should validate identifier

Reply:
@evilebottnawi Could you please elaborate a bit on that? Validate the identifier how?

Reply:
Using regex, for example you can give `if` or `async` name and it will syntax error

Reply:
@evilebottnawi Something like this?
```
// Check for dashes in plugin name
            if(!base.match(/-/g)){ // Regex checks for dashes globally in base
                throw new SyntaxError("Dashes(-) not allowed in ModuleFederationPlugin.name");
            }
```

Reply:
You want to use this regexp: https://stackoverflow.com/a/9392578/2260535

Reply:
@sokra I tried implementing using this regex, and it succeeds in validating the identifier name and check for _illegal characters and reserved keywords_, so `base` is always valid. However the regex to match this is [extremely long](https://stackoverflow.com/a/2008353/14212698) and should not be placed in the `AssignLibraryPlugin.js`. Where should I place this regex and then export it to match for valid identifier name?